elitzur-avro/src/main/scala/com/spotify/elitzur/converters/avro/dynamic/DynamicValidationCompanion.scala

```diff
@@ -0,0 +1,30 @@
+/*
+ * Copyright 2021 Spotify AB.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package com.spotify.elitzur.converters.avro.dynamic
+
+import com.spotify.elitzur.validators.{BaseCompanion, BaseValidationType, Validator}
+
+abstract case class DynamicValidationCompanion(validatorIdentifier: String) {
+  val validator: Validator[_]
+  def dynamicParser: Any => Any
+}
+
+object DynamicValidationCompanionImplicits {
```

Comment:
Is there a strong case to have this as its own implicit? Is it because you don't want to pollute the BaseCompanion type with dynamic info?

Why do we need dynamic parser instead of reusing/repurposing the existing base companion functions?

Reply:
Yes, that's exactly it; wanted to avoid polluting the BaseCompanion type with dynamic info. If that's the only motivation, would you recommend just simply adding this method directly to the BaseCompanion type? If we go with the route, _**I think**_ I can use existing generated mappings in elitzur-validation-types and not have to worry about writing new jinja template code.

Reply:
IMO we already have some stuff that leaks in the companion trait. `bigQueryType` is one such example https://github.com/spotify/elitzur/blob/master/elitzur-core/src/main/scala/com/spotify/elitzur/validators/Companion.scala#L38

The implicit function pattern will be more correct and neater code-wise but I think I'd rather have everything self contained just for readability. This is not a super strong opinion though, there's tonnes of other implicit complexity already existing.

Reply:
Makes sense, thanks! For some reason, I had an impression that the implicit would be better based on the existing uses of implicits, but, yea, code-readability all the way!